File: src/main/java/com/TIComoApp/TIComo/model/Rider.java
package com.TIComoApp.TIComo.model;

import org.springframework.data.mongodb.core.mapping.Document;

import com.mongodb.lang.NonNull;

import lombok.Data;

@Data
@Document(collection = "riders")
public class Rider extends Usuario{

	@NonNull
	private String NIF;
	@NonNull
	private String tipoVehiculo;
	private String matricula;
	private String carnet;
	public Rider(String id, String nombre, String apellidos, String email, String password, String NIF,
			String tipoVehiculo, String matricula, String carnet) {
		super(id, nombre, apellidos, email, password);
		this.NIF = NIF;
		this.tipoVehiculo = tipoVehiculo;
		this.matricula = matricula;
		this.carnet = carnet;
	}
	public String getNIF() {
		return NIF;
	}
	public void setNIF(String NIF) {
		this.NIF = NIF;
	}
	public String getTipoVehiculo() {
		return tipoVehiculo;
	}
	public void setTipoVehiculo(String tipoVehiculo) {
		this.tipoVehiculo = tipoVehiculo;
	}
	public String getMatricula() {
		return matricula;
	}
	public void setMatricula(String matricula) {
		this.matricula = matricula;
	}
	public String getCarnet() {
		return carnet;
	}
	public void setCarnet(String carnet) {
		this.carnet = carnet;
	}
	@Override
	public String toString() {
		return "Rider [NIF=" + NIF + ", tipoVehiculo=" + tipoVehiculo + ", matricula=" + matricula + ", carnet="
				+ carnet + "]";
	}
		
}

File: frontend/webapp/cliente/src/app/models/Rider.ts
import { Usuario } from "./Usuario";

export class Rider extends Usuario{
    /*private*/ nif: String;
    /*private*/ tipoVehiculo: String;
    /*private*/ matricula: String;
    /*private*/ carnet: String;

    
    
    constructor(Nombre: String, Apellidos: String, NIF: String,TipoVehiculo: String,Matricula: String,Carnet: String, Email: String, Password: String){
		super(Nombre,Apellidos,Email,Password);
        this.nif = NIF;
        this.tipoVehiculo = TipoVehiculo;
        this.matricula = Matricula;
        this.carnet = Carnet;
        
    }
	getNIF(): String {
		return this.nif;
	}
	getTipoVehiculo(): String {
		return this.tipoVehiculo;
	}
	getMatricula(): String {
		return this.matricula;
	}
	getCarnet(): String {
		return this.carnet;
	}


}